tests/llmcompressor/transformers/sparsification/test_compress_tensor_utils.py

```diff
@@ -156,60 +158,71 @@ def test_quant_model_reload(format, dtype, tmp_path):
     dataset = "open_platypus"
     concatenate_data = False
     num_calibration_samples = 64
-    output_dir = tmp_path / "oneshot_out"
     splits = {"calibration": "train[:10%]"}
+    empty_model = AutoModelForCausalLM.from_pretrained(model_path, torch_dtype=dtype)
 
     # create a quantized model
     oneshot(
         model=model_path,
         dataset=dataset,
-        output_dir=output_dir,
         num_calibration_samples=num_calibration_samples,
         recipe=recipe_str,
         concatenate_data=concatenate_data,
         splits=splits,
         oneshot_device=device,
+        clear_sparse_session=False,
         precision=dtype,
     )
 
-    model = SparseAutoModelForCausalLM.from_pretrained(
-        tmp_path / "oneshot_out", torch_dtype=dtype
-    )
+    # Fetch the oneshot model
+    model = get_session_model()
+    og_state_dict = model.state_dict()
+    path = tmp_path / "compressed"
 
     for _, module in model.named_modules():
         if hasattr(module, "quantization_scheme"):
             assert module.weight.dtype == dtype
             assert module.quantization_status == QuantizationStatus.FROZEN
 
+    # Save to disk
     model.save_pretrained(
-        tmp_path / "compress_out",
+        path,
         quantization_format=format,
         save_compressed=True,
     )
 
-    config = AutoConfig.from_pretrained(tmp_path / "compress_out")
+    # Verify config on disk
+    config = AutoConfig.from_pretrained(path)
     compression_config = getattr(config, QUANTIZATION_CONFIG_NAME, None)
     quant_config = ModelCompressor.parse_quantization_config(compression_config)
     assert quant_config["format"] == format
 
+    # As HFQuantizer doesn't decompress the model, use the compressor to decompress
+    # the model instead
+    compressor = ModelCompressor.from_compression_config(compression_config)
+    compressor.quantization_config.quantization_status = QuantizationStatus.FROZEN
+    compressor.decompress(model_path=path, model=empty_model)
+
+    # eventually use this pathway once HFQuant Decompression works
+    """
     dense_model = SparseAutoModelForCausalLM.from_pretrained(
-        tmp_path / "compress_out", torch_dtype="auto"
+        "compress_out", torch_dtype="auto", device_map=device
     )
-
-    og_state_dict = model.state_dict()
-    reconstructed_state_dict = dense_model.state_dict()
+    """
+    # Verify the abs difference between the decompressed model
+    # and the original model
+    reconstructed_state_dict = empty_model.state_dict()
     assert len(og_state_dict) == len(reconstructed_state_dict)
     for key in og_state_dict.keys():
-        dense_tensor = og_state_dict[key]
-        reconstructed_tensor = reconstructed_state_dict[key]
+        dense_tensor = og_state_dict[key].to(device)
+        reconstructed_tensor = reconstructed_state_dict[key].to(device)
```

Comment:
Just for my own education, shouldn't both of these models (and their state dicts) be on the cpu?

Reply:
they can be on either the gpu or cpu, as long as it's the same device. cpu offloading wouldnt really be a case here considering the size of the model.